server/src/main/java/org/apache/druid/server/metrics/MetricsModule.java

```diff
@@ -170,13 +174,41 @@ public JvmThreadsMonitor getJvmThreadsMonitor(DataSourceTaskIdHolder dataSourceT
   @Provides
   @ManageLifecycle
   public SysMonitor getSysMonitor(
-      DataSourceTaskIdHolder dataSourceTaskIdHolder
+      DataSourceTaskIdHolder dataSourceTaskIdHolder,
+      Injector injector
   )
   {
+    final Set<NodeRole> nodeRoles = getNodeRoles(injector);
     Map<String, String[]> dimensions = MonitorsConfig.mapOfDatasourceAndTaskID(
         dataSourceTaskIdHolder.getDataSource(),
         dataSourceTaskIdHolder.getTaskId()
     );
-    return new SysMonitor(dimensions);
+    return new SysMonitor(dimensions, isPeonRole(nodeRoles));
```

Comment:
Instead of leaking `isPeon` into the SysMonitor - have you considered introducing a new class `NoopSysMonitor` that does nothing. Then in this provider  if the role is a Peon, return the noop class.

I think that would be a cleaner division of responsibilities and it would make it easier to test as well.

Reply:
I completely agree although I was a little skeptical to use that if any extra functionality might be added in SysMonitor in the future that might want to be used on Peons. But that's a long shot. I'll go with `NoopSysMonitor`. Thanks.